### PR TITLE
Remove -Xlint exclusions in all plugins.

### PR DESCRIPTION
**Pull request by martijnvg**
The xlint exclusions of the following plugins were removed:
* ingest-attachment.
* mapper-size.
* transport-nio. Removing the -try exclusion required some work, because
  the NettyAdaptor implements AutoCloseable and NettyAdaptor#close() method
  could throw an InterruptedException (from ChannelFuture#await() and a generic
  Exception is re-thrown, which maybe an InterruptedException). The easiest way
  around this to me seemed that NettyAdaptor should not implement AutoCloseable,
  because it is not directly used in a try-with-resources statement. @tbrooks8 wdyt?

Relates to #40366

### CODE REVIEW

File: plugins/mapper-size/build.gradle [hunk 1]
```diff
@@ -21,7 +21,3 @@ esplugin {
   description 'The Mapper Size plugin allows document to record their uncompressed size at index time.'
   classname 'org.elasticsearch.plugin.mapper.MapperSizePlugin'
 }
-
-// TODO: migrate to points
```
Comment (martijnvg) [inline plugins/mapper-size/build.gradle]: this todo is no longer relevant, as size field mapper is using points.

**Review by alpar-t — APPROVED** [review]
LGTM

**Comment by martijnvg** [comment]
run elasticsearch-ci/bwc

**Comment by tbrooks8** [comment]
The `AutoCloseable` change looks fine to me.

**Pull request merged**